packages/web-components/fast-foundation/src/data-grid/data-grid-row.ts

```diff
@@ -0,0 +1,293 @@
+import {
+    attr,
+    FASTElement,
+    html,
+    RepeatBehavior,
+    RepeatDirective,
+    observable,
+    ViewTemplate,
+} from "@microsoft/fast-element";
+import {
+    keyCodeArrowLeft,
+    keyCodeArrowRight,
+    keyCodeEnd,
+    keyCodeHome,
+} from "@microsoft/fast-web-utilities";
+import { ColumnDefinition } from "./data-grid";
+
+/**
+ * Enumerates possible row types
+ *
+ * @public
+ * @beta
+ */
+export enum DataGridRowTypes {
+    default = "default",
+    header = "header",
+    stickyHeader = "sticky-header",
+}
+
+/**
+ * A Data Grid Row Custom HTML Element.
+ *
+ * @public
+ */
+export class DataGridRow extends FASTElement {
+    private static cellQueryString =
+        '[role="cell"], [role="gridcell"], [role="columnheader"]';
```

Comment:
is whitespace needed?

Reply:
In the role names?  These are the actual names - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Gridcell_Role

Reply:
Oh, sorry, read that completely backwards.  yes I can remove the spaces.